src/rcore.c

```diff
@@ -3907,6 +3907,40 @@ int GetTouchPointCount(void)
     return CORE.Input.Touch.pointCount;
 }
 
+// Check if a Touch point is pressed once (if the finger touch got in contact with the the screen once, a.k.a TAP, also found in gestures)
+bool IsTouchPointPressed(int index)
+{
+    // Compare the previous and current Touch point state
+    if ((CORE.Input.Touch.currentTouchState[index] == 1) && (CORE.Input.Touch.previousTouchState[index] == 0)) return true;
+
+    return false;
+}
+
+// Check if a Touch point has been released once (if the finger touch was in contact with the screen and now is not anymore)
+bool IsTouchPointReleased(int index)
+{
+    // Compare the previous and current Touch point state
+    if ((CORE.Input.Touch.currentTouchState[index] == 0) && (CORE.Input.Touch.previousTouchState[index] == 1)) return true;
+
+    return false;
+}
+
+// Check if a Touch Point is being pressed (if the finger touch is in contact with the screen)
+bool IsTouchPointDown(int index)
+{
+    // Get the state of the Touch point
+    if (CORE.Input.Touch.currentTouchState[index] == 1) return true;
+
+    return false;
+}
```

Comment:
why the if and not just return the test result?

Reply:
I was following the code syntax in the Mouse events, something to make code easier to read for future users, should I just return `CORE.Input.Touch.currentTouchState[index]` ? I mean the state can be something different in the future or something.

Reply:
Yeah, I usually kept the `result` as a separate variable for readibility...

Reply:
So

```C
if ((CORE.Input.Touch.currentTouchState[index] == 0) && (CORE.Input.Touch.previousTouchState[index] == 1)) return true;

return false;
```

can simply be

``` C
return (CORE.Input.Touch.currentTouchState[index] == 0) && 
           (CORE.Input.Touch.previousTouchState[index] == 1);
```
which is the minimalist solution, since the expression produces the correct truth (**bool**) value.  Alternatively, you can do

```
result = (CORE.Input.Touch.currentTouchState[index] == 0) && 
             (CORE.Input.Touch.previousTouchState[index] == 1);
return result;
```

having declared *result* to be the same type specified for the **function** result.

I prefer the minimalisms and you'll find them in my own code.  Here it is more important to follow the standard pattern preferred by Ramon.